### ISSUE TITLE
Make `TrustKit` dependency explicit

### ISSUE THREAD

**Issue by leviathan** [mapbox/mapbox-events-ios#41]
`MapboxMobileEvents` has an implicitly dependency on `TrustKit` by including it in `/vendor/TrustKit`. When using `TrustKit` in my `Cartfile`:

`Cartfile`
```
github "datatheorem/trustkit" ~> 1.5.0
github "mapbox/mapbox-navigation-ios" ~> 0.16
```

This results in the following compiler warning:

```
error: libarclite_iphonesimulator.a(arclite.o) failed to load objfile for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a

objc[18954]: Class TSKReportsRateLimiter is implemented in both

/Users/.../Library/Developer/CoreSimulator/Devices/76C5F9E7-11F0-4203-8B68-D1BD3367673B/data/Containers/Bundle/Application/5058A930-C33E-47DF-B49E-4F024C50C56F/XY.app/Frameworks/TrustKit.framework/TrustKit (0x1087b8768) 

and 

/Users/.../Library/Developer/CoreSimulator/Devices/76C5F9E7-11F0-4203-8B68-D1BD3367673B/data/Containers/Bundle/Application/5058A930-C33E-47DF-B49E-4F024C50C56F/XY.app/Frameworks/MapboxMobileEvents.framework/MapboxMobileEvents (0x10760a2c0). 

One of the two will be used. Which one is undefined.
```

Could you make the `TrustKit` dependency explicit?
Maybe by adding it to the `MapboxMobileEvents` own Cartfile.

**Comment by simonseyer** [comment]
I'm facing the same issue. This actually leads to rather random runtime crashes because which implementation is used is not defined:

> Class TSKReportsRateLimiter is implemented in both `TrustKit.framework/TrustKit` and `MapboxMobileEvents.framework/MapboxMobileEvents`. One of the two will be used. Which one is undefined.

(seeing many of them when launching my app)

I saw that you also made changes to your copy of the TrustKit framework — this leads to the situation that I cannot even switch to a different TrustKit version because there is no compatible version available. 

I would appreciate if you could switch to an official TrustKit release and add this dependency to the Cartfile. Thanks in advance.

**Comment by bsudekum** [comment]
@rclee I think this could be partially resolved if TrustKit was not vendored here: https://github.com/mapbox/mapbox-events-ios/tree/master/vendor/TrustKit

Do you know the reason we vendored it instead of adding it as a dependency?

**Comment by simonseyer** [comment]
Yeah, that's what I actually wanted to suggest above.

**Comment by friedbunny** [comment]
> Do you know the reason we vendored it instead of adding it as a dependency?

I’d imagine TrustKit is checked in so that gl-native doesn’t have to be aware of this project’s dependency manager(s).

**Comment by rclee** [comment]
@bsudekum We had vendored TrustKit late last year due to some packaging issues. I'll take a closer look into this and see what may be possible to resolve these crashes.

**Comment by simonseyer** [comment]
I was able to work around the crash. The problem was not the duplicate lib (even though this should definitely also be solved IMO). 

Instead, I think the following commit is the problem: https://github.com/mapbox/mapbox-events-ios/commit/0ec259ddc2142ea47fb5f1b9e7f2c5b340bf140a#diff-0f3841af6478cd7d8714e7e3c4f1e56f

Here it is expected, that all domains are pinned and that a pinning configuration is available at all times. However, in our case, we got our own configuration in our Info.plist and this one obviously doesn't include the Mapbox server. This leads to an assertion failure, which only got visible after I integrated the Mapbox Events repo directly into our project. 

The workaround is to actually add the Mapbox pin into our configuration.

```xml
<dict>
	<key>TSKPinnedDomains</key>
	<dict>
		<key>cobi.bike</key>
		<dict>
			our configuration...
		</dict>
		<key>events.mapbox.com</key>
		<dict>
			<key>TSKIncludeSubdomains</key>
			<true/>
			<key>TSKPublicKeyAlgorithms</key>
			<array>
				<string>TSKAlgorithmRsa2048</string>
			</array>
			<key>TSKPublicKeyHashes</key>
			<array>
				<string>yGp2XoimPmIK24X3bNV1IaK+HqvbGEgqar5nauDdC5E=</string>
				<string>fake-key to fix another assertion which requires a backup key</string>
			</array>
			<key>kTSKEnforcePinning</key>
			<true/>
		</dict>
	</dict>
</dict>
```

**Comment by simonseyer** [comment]
As a quick fix removing the following block should also solve the issue: https://github.com/mapbox/mapbox-events-ios/blob/master/vendor/TrustKit/TrustKit.m#L245 It loads the Info.plist when available and this way prevents the library from getting initialized with the configuration defined in code.

**Comment by alfwatt** [comment]
For Carthage we can add TrustKit to the `Cartfile`
For CocoaPods we can add TrustKit to the `podspec`
For other build systems we can add `components.sh` and some `Makefile` directives

**Comment by alfwatt** [comment]
This will be fixed in 0.9.3 by removing the dependance on TrustKit from the framework

**Comment by alfwatt** [comment]
Fixed: #126
